src/dosbox.cpp

```diff
@@ -361,6 +362,8 @@ static void DOSBOX_RealInit(Section * sec) {
 	Section_prop * section=static_cast<Section_prop *>(sec);
 	/* Initialize some dosbox internals */
 
+	use_delay_precise = CanDelayPrecise();
```

Comment:
Did a quick test replacing the branches with a one-time assignment to a `std::function` object. 

``` diff

--- a/include/timer.h
+++ b/include/timer.h
@@ -24,6 +24,7 @@
 #include <cstdlib>
 
 #include <chrono>
+#include <functional>
 #include <limits>
 #include <thread>
 
@@ -75,7 +76,9 @@ static inline int GetTicksUsSince(int64_t old_ticks)
        return GetTicksDiff(now, old_ticks);
 }
 
-static inline void Delay(int milliseconds)
+extern std::function<void(int)> Delay;
+
+static inline void DelayCoarse(int milliseconds)
 {
        std::this_thread::sleep_for(std::chrono::milliseconds(milliseconds));
 }
diff --git a/src/dosbox.cpp b/src/dosbox.cpp
index d3095cef..93be6fbc 100644
--- a/src/dosbox.cpp
+++ b/src/dosbox.cpp
@@ -135,7 +135,6 @@ static LoopHandler * loop;
 
 bool SDLNetInited;
 
-static bool use_delay_precise = false;
 static int ticksRemain;
 static int64_t ticksLast;
 static int ticksAdded;
@@ -145,6 +144,8 @@ bool ticksLocked;
 void increaseticks();
 bool mono_cga=false;
 
+std::function<void(int)> Delay = {};
+
 static Bitu Normal_Loop()
 {
        Bits ret;
@@ -195,7 +196,7 @@ void increaseticks() { //Make it return ticksRemain and set it in the function a
                ticksAdded = 0;
 
                if (!CPU_CycleAutoAdjust || CPU_SkipCycleAutoAdjust || sleep1count < 3) {
-                       use_delay_precise ? DelayPrecise(1) : Delay(1);
+                       Delay(1);
                } else {
                        /* Certain configurations always give an exact sleepingtime of 1, this causes problems due to the fact that
                           dosbox keeps track of full blocks.
@@ -204,7 +205,7 @@ void increaseticks() { //Make it return ticksRemain and set it in the function a
                        static const Bit32u sleeppattern[] = { 2, 2, 3, 2, 2, 4, 2 };
                        static Bit32u sleepindex = 0;
                        if (ticksDone != lastsleepDone) sleepindex = 0;
-                       use_delay_precise ? DelayPrecise(sleeppattern[sleepindex++]) : Delay(sleeppattern[sleepindex++]);
+                       Delay(sleeppattern[sleepindex++]);
                        sleepindex %= sizeof(sleeppattern) / sizeof(sleeppattern[0]);
                }
                auto timeslept = GetTicksSince(ticksNew);
@@ -362,7 +363,7 @@ static void DOSBOX_RealInit(Section * sec) {
        Section_prop * section=static_cast<Section_prop *>(sec);
        /* Initialize some dosbox internals */
 
-       use_delay_precise = CanDelayPrecise();
+       Delay = CanDelayPrecise() ? DelayPrecise : DelayCoarse;
 
        ticksRemain=0;
        ticksLast=GetTicks();
```